spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/security/OAuthSecurityConfiguration.java

```diff
@@ -41,46 +65,127 @@
+			.httpBasic().and()
+			.logout()
+			.logoutSuccessUrl(dashboard("/logout-success-oauth.html"))
+			.and().csrf().disable()
+			.exceptionHandling()
+			.defaultAuthenticationEntryPointFor(new LoginUrlAuthenticationEntryPoint("/login"), textHtmlMatcher)
+			.defaultAuthenticationEntryPointFor(basicAuthenticationEntryPoint, AnyRequestMatcher.INSTANCE);
 
 		securityStateBean.setAuthenticationEnabled(true);
 		securityStateBean.setAuthorizationEnabled(false);
 	}
 
 	@Bean
+	public UserInfoTokenServices tokenServices() {
+		final UserInfoTokenServices tokenServices = new UserInfoTokenServices(resourceServerProperties.getUserInfoUri(),
+				authorizationCodeResourceDetails.getClientId());
+		tokenServices.setRestTemplate(oAuth2RestTemplate());
+		return tokenServices;
+
+	}
+
+	@Bean
+	public OAuth2RestTemplate oAuth2RestTemplate() {
+		final OAuth2RestTemplate oAuth2RestTemplate = new OAuth2RestTemplate(authorizationCodeResourceDetails, oauth2ClientContext);
+		return oAuth2RestTemplate;
+	}
+
+	@Bean
 	public AuthenticationProvider authenticationProvider() {
 		return new ManualOAuthAuthenticationProvider();
 	}
 
+	@Bean
+	public ProviderManager providerManager() {
+		List<AuthenticationProvider> providers = new ArrayList<>();
+		providers.add(this.authenticationProvider());
+		ProviderManager providerManager = new ProviderManager(providers);
+		return providerManager;
+	}
+
+	private Filter oauthFilter() {
+		final OAuth2ClientAuthenticationProcessingFilter oauthFilter = new OAuth2ClientAuthenticationProcessingFilter(
+				"/login");
+		oauthFilter.setRestTemplate(oAuth2RestTemplate());
+		oauthFilter.setTokenServices(tokenServices());
+		return oauthFilter;
+	}
+
 	private OAuth2AuthenticationProcessingFilter oAuth2AuthenticationProcessingFilter() {
 		final OAuth2AuthenticationProcessingFilter oAuth2AuthenticationProcessingFilter = new OAuth2AuthenticationProcessingFilter();
 		oAuth2AuthenticationProcessingFilter.setAuthenticationManager(oauthAuthenticationManager());
 		oAuth2AuthenticationProcessingFilter.setStateless(false);
 		return oAuth2AuthenticationProcessingFilter;
 	}
 
-	private AuthenticationManager oauthAuthenticationManager() {
+	@Bean
+	public AuthenticationManager oauthAuthenticationManager() {
 		final OAuth2AuthenticationManager oauthAuthenticationManager = new OAuth2AuthenticationManager();
-		oauthAuthenticationManager.setTokenServices(tokenServices);
+		oauthAuthenticationManager.setTokenServices(tokenServices());
 		return oauthAuthenticationManager;
 	}
+
+	private static class BrowserDetectingContentNegotiationStrategy extends HeaderContentNegotiationStrategy {
+
+		@Override
+		public List<MediaType> resolveMediaTypes(NativeWebRequest request)
+				throws HttpMediaTypeNotAcceptableException {
+			final List<MediaType> supportedMediaTypes = super.resolveMediaTypes(request);
+
+			final String userAgent = request.getHeader(HttpHeaders.USER_AGENT);
+			if (userAgent != null && userAgent.contains("Mozilla/5.0")
```

Comment:
Why get rid of basic auth?  We can't assume OAuth setup everywhere, especially just 'out of the box' where someone on PWS/PCF or k8s on google would expose an app to the public without any security....

Merging now though....